### ISSUE TITLE
Ltac2 expose equality on inductive and constructor?

### ISSUE THREAD

**Issue by gmalecha** [coq/coq#12786]
#### Description of the problem

This is a feature request. The types `constructor` and `inductive` are exposed in the `Unsafe` module, but (as far as I can tell) there are no real eliminators for these types. Would it be easy to expose equality on these types? That would allow compiling pattern matching to finer-grained operations.

#### Coq Version

```
The Coq Proof Assistant, version 8.11.2 (July 2020)
compiled on Jul 26 2020 0:08:43 with OCaml 4.07.1
```

**Comment by ppedrot** [comment]
Half of this issue is fixed by #13920.

**Comment by SkySkimmer** [comment]
Done in https://github.com/coq/coq/pull/16537
